test/Spec/React18HooksSpec.purs

```diff
@@ -0,0 +1,143 @@
+module Test.Spec.React18HooksSpec where
+
+import Prelude
+
+import Control.Monad.Rec.Class (forever)
+import Data.Array as Array
+import Data.Foldable (for_, traverse_)
+import Data.Maybe (fromMaybe)
+import Data.Monoid (guard, power)
+import Data.String as String
+import Data.Tuple.Nested ((/\))
+import Effect.Aff (Milliseconds(..), apathize, delay, launchAff_)
+import Effect.Class (liftEffect)
+import Effect.Ref as Ref
+import Foreign.Object as Object
+import React.Basic (fragment)
+import React.Basic.DOM as R
+import React.Basic.DOM.Events (targetValue)
+import React.Basic.Events (handler, handler_)
+import React.Basic.Hooks (reactComponent)
+import React.Basic.Hooks as Hooks
+import React.TestingLibrary (cleanup, fireEventClick, renderComponent, typeText)
+import Test.Spec (Spec, after_, before, describe, it)
+import Test.Spec.Assertions (shouldNotEqual)
+import Test.Spec.Assertions.DOM (textContentShouldEqual)
+import Web.DOM.Element (getAttribute)
+import Web.HTML.HTMLElement as HTMLElement
+
+spec ∷ Spec Unit
+spec =
+  after_ cleanup do
+    before setup do
+      describe "React 18 hooks" do
+        it "useId works" \{ useId } -> do
+          { findByTestId } <- renderComponent useId {}
+          elem <- findByTestId "use-id"
+          idʔ <- getAttribute "id" (HTMLElement.toElement elem) # liftEffect
```

Comment:
😮 I never knew you could use `?` in identifiers........

Reply:
Oh, it's a habit, I can remove it. It's not `?` but `ʔ` which is from [a glottal stop](https://en.wikipedia.org/wiki/Glottal_stop_(letter))

Reply:
```json
  "Almost question mark": {
		"scope": "purescript",
		"prefix": "?",
		"body": [
      "ʔ"
		],
		"description": "Append a ʔ to a variable name which is an allowed identifier"
  },
```
I have that in a snippets file so I can do question mark tab to get it.